frontend/src/components/Login/Login.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react'
+import axios from 'axios'
+import { Link, useLocation, useNavigate } from 'react-router-dom'
+import Layout from '../Layout/Layout'
+import useAuth from '../../hooks/useAuth'
+import './Login.css'
+
+type Props = {
+   addBikeInputRef: React.MutableRefObject<HTMLInputElement>
+}
+
+function Login(props: Props) {
+   const location = useLocation()
+   const navigate = useNavigate()
+   const currentUser = useAuth(false)
+   !!currentUser && navigate('/')
+   const [username, setUsername] = useState(location.state?.username || '')
```

Comment:
What is `location.state?.username`?

Reply:
[and this](https://developer.mozilla.org/en-US/docs/Web/API/Location?retiredLocale=de)

Reply:
["and this wonderful article" - Gleb ](https://medium.com/frontendweb/how-to-pass-state-or-data-in-react-router-v6-c366db9ee2f4)